resteasy-jaxrs/src/main/java/org/jboss/resteasy/plugins/providers/sse/EventByteArrayOutputStream.java

```diff
@@ -5,23 +5,23 @@
 
 public class EventByteArrayOutputStream extends ByteArrayOutputStream
 {
-   private void removeBlankLine()
+   public synchronized byte[] getEventPayLoad()
    {
-      if (this.count > 4)
+      // delimiter is \r or \n
+      if (count >=2 && this.buf[count-2] == this.buf[count-1])
       {
-         count = count - 4;
+         return Arrays.copyOf(buf, count -1);
+      }
+      //delimiter is \r\n
+      if (count >= 2 && buf[count-2] == '\r' && buf[count-1] == '\n') {
+         Arrays.copyOf(buf, count-2);
```

Comment:
@jimma , as Ron pointed out, isn't a return statement still missing here?

Reply:
Ah. What a mistake ! Fixed. Thanks @ronsigal and @asoldano